Botkit/Microsoft.BotKit.Adapters.Slack/SlackAdapter.cs

```diff
@@ -188,14 +172,36 @@ public object ActivityToSlack(Activity activity)
                 };
                 message.attachments.Add(newAttachment);
             }
-                
+
             message.channel = activity.Conversation.Id;
-            //message.thread_ts = JsonConvert.DeserializeObject<DateTime>(activity.Conversation.Properties["thread_ts"].ToString());
+
+            if (activity.Conversation.Properties["thread_ts"].ToString() != string.Empty)
+            {
+                //object obj = JsonConvert.DeserializeObject<object>(activity.Conversation.Properties["thread_ts"].ToString());
+                //message.ThreadTS = (obj != null) ? (DateTime)obj : default(DateTime);
+                message.ThreadTS = activity.Conversation.Properties["thread_ts"].ToString();
+            }
 
             // if channelData is specified, overwrite any fields in message object
             if (activity.ChannelData != null)
             {
-                //message = activity.ChannelData;
+                try
+                {
+                    // Try a straight up cast
+                    message = activity.ChannelData as NewSlackMessage;
+                }
+                catch (InvalidCastException ex)
+                {
+                    foreach (var property in message.GetType().GetFields())
+                    {
+                        string name = property.Name;
+                        var value = ((activity as dynamic)[name] != null) ? (activity as dynamic)[name] : null;
```

Comment:
I belive this line should be `(activity.ChannelData as dynamic)`

Reply:
true